web/modules/custom/weather/src/Services/WeatherApi.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+namespace Drupal\weather\Services;
+
+use Drupal\Core\Config\ConfigFactoryInterface;
+use GuzzleHttp\ClientInterface;
+
+/**
+ * Service class WeatherApi.
+ */
+class WeatherApi {
+
+  /**
+   * Constructs a ConfigFactory and a Client object.
+   */
+  public function __construct(
+    protected ConfigFactoryInterface $configFactory,
+    protected ClientInterface $httpClient,
+  ) {}
+
+  /**
+   * Weather Api Validation.
+   */
+  public function validateApi(string $apiKey): bool {
+    try {
+      $client = $this->httpClient->get('http_client');
+      $client->get('https://api.openweathermap.org/data/2.5/weather?q=London&appid=' . $apiKey);
+    }
+    catch (\Exception $error) {
+      return FALSE;
+    }
+    return TRUE;
+  }
+
+  /**
+   * Weather City Validation.
+   */
+  public function validateCityApi(string $cityName): bool {
+    $config = $this->configFactory->get('weather.settings');
+    $apiKey = $config->get('api_admin_key');
+    try {
+      $client = $this->httpClient->get('http_client');
+      $res = $client->get('https://api.openweathermap.org/data/2.5/weather?q=' . $cityName . '&units=metric&appid=' . $apiKey);
+      $body = (string) $res->getBody();
+      json_decode($body, TRUE);
```

Comment:
Do you need to get the body if the API returns 404 for the wrong city?
json_decode is most likely doing nothing as well.